upgrades: modify InjectLegacyTable to handle dynamically assigned IDs

andyyang890 opened the pull request:
This patch modifies the `InjectLegacyTable` function, which is used by
upgrade tests to inject an old version of a table descriptor, to handle
the case where the table has a dynamically assigned ID.

Epic: None

Release note: None

andyyang890 committed:
upgrades: modify InjectLegacyTable to handle dynamically assigned IDs

This patch modifies the `InjectLegacyTable` function, which is used by
upgrade tests to inject an old version of a table descriptor, to handle
the case where the table has a dynamically assigned ID.

Release note: None

postamar reviewed (dismissed):
LGTM assuming that this actually works correctly for dynamically-allocated ID system tables (I trust that you've checked)

andyyang890 commented:
Hmm I realized that when I was testing this earlier, I forgot to remove the code that queried and added the table ID on the deprecated descriptor. I think I've fixed it now but I'll make sure it's definitely working before asking for another review.

FYI I'm planning on using this function in #93090 so the test in that PR will test that this change works.

andyyang890 commented:
TFTR!

bors r=postamar

andyyang890 commented:
resolved merge conflict

bors r=postamar
